Raster Code does not compile on desktop

If you try and build the new raster code on the desktop with Intel Fortran and GNU C, you'll get an error:
```
gcc: error: unrecognized command line option ‘-qopenmp’; did you mean ‘-fopenmp’?
make[3]: *** [src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/CMakeFiles/raster.dir/zip.c.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/CMakeFiles/raster.dir/all] Error 2
make[1]: *** [src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/CMakeFiles/mkCatchParam.x.dir/rule] Error 2
make: *** [mkCatchParam.x] Error 2
```
The issue is that the `CMakeLists.txt` is assuming that the OpenMP flags for Fortran and C are the same:
https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/8097811a17b357061ac6ef8e5869f3dcc39ed330/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/CMakeLists.txt#L27
Since `${this}` contains both C and Fortran code, we can't assume this.

The solution is, I think, to change this line to:
```cmake
target_link_libraries(${this} PRIVATE OpenMP::OpenMP_Fortran OpenMP::OpenMP_C)
```
Doing this does let things build, but I do not know how to test it. I will ask @smahanam to try it out as he knows how to run this code.

--- a/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/CMakeLists.txt
+++ b/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/CMakeLists.txt
@@ -24,7 +24,7 @@ include_directories(${INC_ESMF})
 include_directories(${INC_NETCDF})
 
 esma_add_library(${this} SRCS ${srcs} DEPENDENCIES MAPL_Base)
-target_compile_options(${this} PRIVATE ${OpenMP_Fortran_FLAGS})
+target_link_libraries(${this} PRIVATE OpenMP::OpenMP_Fortran OpenMP::OpenMP_C)
 
 if(NOT FORTRAN_COMPILER_SUPPORTS_FINDLOC)
    target_compile_definitions(${this} PRIVATE USE_EXTERNAL_FINDLOC)

